drivers/volume/portworx/portworx.go

```diff
@@ -744,75 +862,174 @@ func (p *portworx) createVolumeSnapshotData(
 	})
 
 	if err != nil {
-		logrus.Errorf("Error creating the VolumeSnapshotData for snap %s: %v", snapshotName, err)
-		return nil, fmt.Errorf("Failed to create the VolumeSnapshotData for snap %s", snapshotName)
+		err = fmt.Errorf("error creating the VolumeSnapshotData for snap %s due to err: %v", snapshotName, err)
+		logrus.Println(err)
+		return nil, err
 	}
 
 	return result, nil
 }
 
 func (p *portworx) findParentPVCBySnapID(snapID string) (*v1.PersistentVolumeClaim, error) {
```

Comment:
This will cause issues since the parent PVC can be deleted after the PX snaps have been taken but before the volumesnapshot objects have been created.